### PR TITLE
Docker optimizations

### PR DESCRIPTION
**Pull request by kikkomep**
This PR introduces a new pure Python implentation of the helper script to launch the dockerized version of wft4galaxy. As the previous version, it can be launched from terminal by typing `wft4galaxy-docker` after wft4galaxy installation or downloaded and installed as a standalone script from the wft4galaxy repository by typing:

    curl -s https://raw.githubusercontent.com/phnmnl/wft4galaxy/develop/utils/docker/install.sh | bash

Until this PR has been merged, use the `docker-optimizations` branch of my repository (i.e., https://github.com/kikkomep/wft4galaxy) to install wft4galaxy.  To download the standalone `wft4galaxy-docker` script in your current working dir, cut and paste the following line to your terminal:

    curl -s https://raw.githubusercontent.com/kikkomep/wft4galaxy/docker-optimizations/utils/docker/install.sh | bash /dev/stdin --repo kikkomep/wft4galaxy --branch docker-optimizations .

and launch `wft4galaxy-docker` with the two options shown below to use the latest Docker images:

    ./wft4galaxy-docker --repository kikkomep --version docker-optimizations [...]

As an example, you can launch the ChangeCase workflow test case as follows:

    ./wft4galaxy-docker --repository kikkomep --version docker-optimizations -f examples/change_case/workflow-test.yml

### CODE REVIEW

File: wft4galaxy/app/wizard.py [hunk 1]
```diff
@@ -28,10 +28,13 @@
...
-_logging.basicConfig(format=LogFormat)
-_logger = _logging.getLogger("WorkflowTest")
-_logger.setLevel(_logging.INFO)
+_logger = _common.default_logger
+
+# fix an issue related to the output buffering
+# when the wizard is running within a Docker container
+if _sys.version_info[0] == 2:
+    _logger.debug("Disabling output buffering...")
+    _sys.stdout = _os.fdopen(_sys.stdout.fileno(), 'w', 0)
```
Comment (ilveroluca) [inline wft4galaxy/app/wizard.py]: Consider wrapping this within a function that only gets called when the operation is relevant, for instance from `main()`.  There are cases where you wouldn't want a module to overwrite `sys.stdout` upon loading, such as when running unit tests.

File: wft4galaxy/common.py [hunk 2]
```diff
@@ -12,17 +12,21 @@
 ENV_KEY_GALAXY_URL = "GALAXY_URL"
 ENV_KEY_GALAXY_API_KEY = "GALAXY_API_KEY"
 
+# configure logger
+_log_format = "%(asctime)s [wft4galaxy] [%(levelname)-5.5s]  %(message)s"
+default_logger = _logging.getLogger("WorkflowTest")
```
Comment (ilveroluca) [inline wft4galaxy/common.py]: I would also wrap these lines (17-21) into a `config_logging()` function that gets called when the application is started, not when the module is loaded.

File: wft4galaxy/app/wizard.py (hunk 1 above)
Comment (kikkomep) [reply wft4galaxy/app/wizard.py]: Ok! Fixed via https://github.com/phnmnl/wft4galaxy/pull/18/commits/109f144e981e4b852c46b1b8aa4c2b1ff40bf8a0

File: wft4galaxy/common.py (hunk 2 above)
Comment (kikkomep) [reply wft4galaxy/common.py]: Fixed via https://github.com/phnmnl/wft4galaxy/pull/18/commits/5120910d58cf0d76a6f4ad8e9a13639efa7919ec

**Review by ilveroluca — APPROVED** [review]

**Pull request merged**